yt/yt/server/query_tracker/query_tracker_proxy.cpp

```diff
@@ -0,0 +1,1164 @@
+            };
+            transaction->WriteRows(
+                StateRoot_ + "/finished_queries",
+                TFinishedQueryDescriptor::Get()->GetNameTable(),
+                MakeSharedRange(std::move(rows), rowBuffer));
+        }
+        {
+            TFinishedQueryByStartTimePartial record{
+                .Key = {.StartTime = *query.StartTime, .QueryId = queryId},
+            };
+            if (options.Annotations) {
+                record.FilterFactors = filterFactors;
+            }
+            if (options.AccessControlObject) {
+                record.AccessControlObject = options.AccessControlObject;
+            }
+            std::vector rows{
+                record.ToUnversionedRow(rowBuffer, TFinishedQueryByStartTimeDescriptor::Get()->GetIdMapping()),
+            };
+            transaction->WriteRows(
+                StateRoot_ + "/finished_queries_by_start_time",
+                TFinishedQueryByStartTimeDescriptor::Get()->GetNameTable(),
+                MakeSharedRange(std::move(rows), rowBuffer));
+        }
+    } else {
+        auto rowBuffer = New<TRowBuffer>();
+        {
+            TActiveQueryPartial record{
+                .Key = {.QueryId = queryId},
+            };
+            if (options.Annotations) {
+                record.Annotations = ConvertToYsonString(options.Annotations);
+                record.FilterFactors = GetFilterFactors(record);
+            }
+            if (options.AccessControlObject) {
+                record.AccessControlObject = options.AccessControlObject;
+            }
+            std::vector rows{
+                record.ToUnversionedRow(rowBuffer, TActiveQueryDescriptor::Get()->GetIdMapping()),
+            };
+            transaction->WriteRows(
+                StateRoot_ + "/active_queries",
+                TActiveQueryDescriptor::Get()->GetNameTable(),
+                MakeSharedRange(std::move(rows), rowBuffer));
+        }
+    }
+
+    WaitFor(transaction->Commit())
+        .ThrowOnError();
+}
+
+TGetQueryTrackerInfoResult TQueryTrackerProxy::GetQueryTrackerInfo(
+    const TGetQueryTrackerInfoOptions& options)
+{
+    YT_LOG_DEBUG(
+        "Getting query tracker information (Attributes: %v)",
+        options.Attributes);
+
+    auto attributes = options.Attributes;
+
+    attributes.ValidateKeysOnly();
+
+    TString clusterName = "";
+
+    if (attributes.AdmitsKeySlow("cluster_name")) {
+        YT_LOG_DEBUG("Getting cluster name");
+        clusterName = StateClient_->GetClusterName().value_or("");
+    }
+
+    TNodeExistsOptions nodeExistsOptions;
+    nodeExistsOptions.ReadFrom = EMasterChannelKind::Cache;
+
+    static const TYsonString EmptyMap = TYsonString(TString("{}"));
+    TYsonString supportedFeatures = EmptyMap;
+    if (attributes.AdmitsKeySlow("supported_features")) {
+        supportedFeatures = BuildYsonStringFluently()
+            .BeginMap()
+                .Item("access_control").Value(true)
+            .EndMap();
```

Comment:
Would be nice to do it like here: https://github.com/ytsaurus/ytsaurus/blob/45a5e988eaed39c6b9790b094f6ad5dbf7d6ff6d/yt/yt/client/driver/etc_commands.cpp#L78-L81
With a comment "These features are guaranteed to be deployed before or with this code."